### DISCUSSION TITLE
How to add JSLint toAtom and VSCode on Windows?

### DISCUSSION THREAD

**Discussion by kaizhu256** [jslint-org/jslint#357] [Q&A] [answered]
> Hello Douglas,
>             I read the instructions on GitHub of JSLint, but couldn't find how to add JSLint to Atom and VSCode and other editors on Windows. The Instructions were available only for Vim. Can you please instruct me on how to add JSLint to Atom and other editors on Windows?
> 
> Varuna

latest jslint currently has no official plugin-support for atom or vscode.

for older versions of jslint, there are examples of how to do so @ https://github.com/ajhyndman/vscode-jslint/tree/master/jslint and https://github.com/tcarlsen/atom-jslint.

i think adding support for vscode would be good for jslint's visibility (its 3rd most popular ide @ https://pypl.github.io/IDE.html).
in future, might fork the above examples and get them running with latest jslint.

**Comment by mauskin** [comment]
I have a VS Code plugin that ran JSLint 2019-09-17 edition on v.1.39. I stopped using it after I switched from VS Code.

Do you need help with it?

**Reply by kaizhu256** [reply]
if you post a github repo with working plugin, i can take a look at it. some thoughts:

- i might modify the plugin to disable auto-lint.  the jslint-vim-plugin was originally auto-linting, but experience showed it overloaded the user (me) with too many warnings when i open random js-files for editing -- thus feature was removed.
- being unfamiliar with vscode, is there a way to programmatically bind hotkeys (e.g. `ctrl-s + ctrl-j`) to jslint, so user can selectively lint files in a web-project?

**Reply by mauskin** [reply]
I’ll see if I can make it run on latest JSLint on latest VS Code and create a repo.

**Comment by kaizhu256** [comment, marked as answer]
fyi, pr #395 added vscode-support for latest jslint (v2022.5.1-beta).

# Quickstart JSLint in VS Code


<br><br>
### To run JSLint in VS Code:
1. In VS Code, search and install extension [`vscode-jslint`](https://marketplace.visualstudio.com/items?itemName=jslint.vscode-jslint)
2. In VS Code, while editing a javascript file:
    - right-click context-menu and select `[JSLint - Lint File]`
    - or use key-binding `[Ctrl + Shift + J], [L]`
    - or use key-binding `[ Cmd + Shift + J], [L]` for Mac
- screenshot

[![screenshot](https://user-images.githubusercontent.com/280571/166092843-bcb5ebfb-748e-4fc3-9380-df11525ec9ce.png)](https://marketplace.visualstudio.com/items?itemName=jslint.vscode-jslint)
